### PR TITLE
Subscriptions are message feeds

### PR DESCRIPTION
**Pull request by mhidalgo-bdai**
This patch generalizes the `Subscription` API, in preparation for other message streams to come (synchronized, filtered by transform availability).

### CODE REVIEW

File: bdai_ros2_wrappers/bdai_ros2_wrappers/action.py [hunk 1]
```diff
@@ -358,7 +358,8 @@ def __init__(self, action_type: Type, action_name: str, node: Optional[Node] = N
             kwargs: other keyword arguments are forwarded to the underlying action client.
             See `rclpy.action.ActionClient` documentation for further reference.
         """
-        node = node or scope.node()
+        if node is None:
```
Comment (khughes-bdai) [inline bdai_ros2_wrappers/bdai_ros2_wrappers/action.py:361]: Is some of this a duplicate of https://github.com/bdaiinstitute/ros_utilities/pull/97 ?

**Review by khughes-bdai — APPROVED** [review]
LGTM

File: bdai_ros2_wrappers/bdai_ros2_wrappers/action.py (hunk 1 above)
Comment (mhidalgo-bdai) [reply bdai_ros2_wrappers/bdai_ros2_wrappers/action.py:361]: Indeed. I have to rebase now that #97 got in.

**Comment by mhidalgo-bdai** [comment]
Rebased!

**Pull request merged**